### DISCUSSION TITLE
getStaticProps need custom context parameter

### DISCUSSION THREAD

**Discussion by Yuiffy** [vercel/next.js#37879] [Ideas]
### Describe the feature you'd like to request

[Context parameter](https://nextjs.org/docs/api-reference/data-fetching/get-static-props#context-parameter)
 route parameters now only support route param / preview param / i18n param.

I have a website need to support two domains with different content. I want to add a domain param to getStaticProps, make it generate two different ISR page by different domain. 

### Describe the solution you'd like

getStaticProps can add custom parameter

### Describe alternatives you've considered

getStaticProps can add custom parameter

**Comment by icyJoseph** [comment]
### Alternative

One alternative is to create a folder structure like `/_sites/[title]/index.tsx` for example, and then define `getStaticPaths` which returns a list with the supported titles, and that in turn is passed to the `getStaticProps` function.

Now, the issue is that you'd need to rewrite requests, so that if a user arrives to `example.com` you rewrite them to `/_sites/example/index.tsx`, the user URL still shows `example.com`. You could do this in middleware, but the last issue there would be, how do you decide the domain that you are serving? Most check the `host` header, but that header, under certain circumstances, could be missing.

How did you plan to figure out which domain you are serving? 

Have you considered, using different .env var files for each domain, and based on the values of those you render differently, after all, two domains is not terribly bad.
